Develop

S8ight opened the pull request:
Completely finished the layout of the page, but still needed to connect the Shopify API for loading products (now they are loaded from a json file).

S8ight committed:
Finished the layout of the category page, made it responsive

S8ight committed:
Minor changes in styles for better responsive view

AlexMartin06 commented on pages/categoryPage/category.html:
```diff
@@ -2,10 +2,478 @@
 </head>
 <body>
+<div class="header">
+    <div class="hamburgerMenu">
```
all class names should be: class="example-class-name"

S8ight replied:
> all class names should be: class="example-class-name"

You can't use camelCase for classes, or kebab-case is simply just a better practice?

AlexMartin06 commented on pages/categoryPage/category.html:
```diff
@@ -2,10 +2,478 @@
+    <div class="filtersContainer">
+
+        <div class="filterBlock">
+            <p>CATEGORIES</p>
```
use text-transform instead of instant uppercase

AlexMartin06 commented on pages/categoryPage/category.html:
```diff
@@ -2,10 +2,478 @@
+        <div class="filterBlock">
+            <p>CATEGORIES</p>
+            <ul>
+                <li><button onclick="openSubList('Arrivals')">
```
add 'id' for button selector and after this add event listener in js file using this id

AlexMartin06 commented on pages/categoryPage/category.html:
```diff
@@ -2,10 +2,478 @@
+                    <img class="filterIcon" src="../../assets/icons/angle-down.svg" alt="arrow"/>
+                </button>
+
+                    <div class="categoriesSubList" id="Accessories">
```
id starts from small letter
also use this template: id="test-id"

AlexMartin06 commented on pages/categoryPage/category.html:
```diff
@@ -2,10 +2,478 @@
+                    <input type="checkbox" style="background-color: #fff;" class="colorCheckbox"/>
+                    <input type="checkbox" style="background-color: #000;" class="colorCheckbox"/>
+                    <input type="checkbox" style="background-color: #51a909;" class="colorCheckbox"/>
+                    <input type="checkbox" style="background-color: #497bee;" class="colorCheckbox"/>
```
instead of inline styles add styles in css file

AlexMartin06 commented on pages/categoryPage/category.html:
```diff
@@ -2,10 +2,478 @@
+                    <div class="priceProgress"></div>
+                </div>
+                <div class="rangeInput">
+                    <input type="range" class="rangeMin" min="0" max="500" value="0" step="10">
```
to do: get min and max price from products json and use this in your js file

vdymashok replied:
Also, get color from products json and use this in your js file

AlexMartin06 commented on pages/categoryPage/js/categoryPageScript.js:
```diff
@@ -0,0 +1,122 @@
+const productContainer = document.querySelector('.productCardsContainer');
+const jsonFileURL = '../../assets/products/products.json';
+
+fetch(jsonFileURL)
```
try to separate your logic in different functions

AlexMartin06 commented on pages/categoryPage/category.css:
```diff
@@ -0,0 +1,913 @@
+@import url('https://fonts.googleapis.com/css2?family=Rubik:wght@300;400;500&display=swap');
```
can you add google fonts from head in html file?

S8ight replied:
> can you add google fonts from head in html file?

Sure, I will cange it

vdymashok replied:
I also think would be better If you set the font family for the "body" instead of setting it for each element, but it is not critical

vdymashok commented on pages/categoryPage/js/categoryPageScript.js:
```diff
@@ -0,0 +1,122 @@
+fetch(jsonFileURL)
+    .then(response => response.json())
+    .then(data => {
+        data.forEach(product => {
```
Try using template strings to create product cards

vdymashok replied:
And also, try using css variables for colors

vdymashok commented on pages/categoryPage/category.css:
```diff
@@ -0,0 +1,913 @@
+.productCountInfo p {
+    font-family: 'Rubik', sans-serif;
+    font-size: 0.9em;
+    white-space: nowrap;
```
using white-space: nowrap; this is not a best practice 
In this case, if we have a long category name, it will not fit on the page

S8ight committed:
Changes after code review.

S8ight committed:
Made the last changes after the code review.
Wrote an api to connect to the Shopify api, now products are uploaded from the Shopify store.

S8ight committed:
Minor changes in styles.

S8ight committed:
Minor changes, removed the extra code.

vdymashok commented on pages/categoryPage/js/category-page-script.js:
```diff
@@ -0,0 +1,228 @@
+    const productContainer = document.querySelector('.product-cards-container');
+
+    products.forEach(product => {
+        const productCard = document.createElement('div');
```
Try to use 'template string'
For example:

```js
 `<div class="product-card">
<img src="${product.image}" alt="${product.name}">
...
</div>
`
```

S8ight replied:
Thanks for the example, now I understand what you meant.
Already fixed it.

vdymashok replied:
Great work

S8ight committed:
Rewritten function dynamicProductsLoading with using template string.
